feature/notification/src/main/java/io/github/droidkaigi/confsched2019/notification/worker/ManageTopicSubscriptionWorker.kt

```diff
@@ -0,0 +1,105 @@
+package io.github.droidkaigi.confsched2019.notification.worker
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.Data
+import androidx.work.ExistingWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import com.google.firebase.iid.FirebaseInstanceId
+import com.google.firebase.messaging.FirebaseMessaging
+import io.github.droidkaigi.confsched2019.notification.Topic
+import io.github.droidkaigi.confsched2019.timber.error
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.tasks.await
+import timber.log.Timber
+import timber.log.debug
+
+class ManageTopicSubscriptionWorker(
+    context: Context,
+    parameters: WorkerParameters
+) : Worker(context, parameters) {
+    override fun doWork(): androidx.work.Result {
+        try {
+            if (!inputData.isValid) {
+                return androidx.work.Result.failure()
+            }
+
+            val topicsToBeSubscribed =
+                inputData.getStringArray(KEY_TOPIC_NAMES_TO_BE_SUBSCRIBED).orEmpty()
+
+            val topicsToBeUnsubscribed =
+                inputData.getStringArray(KEY_TOPIC_NAMES_TO_BE_UNSUBSCRIBED).orEmpty()
+
+            runBlocking {
+                FirebaseInstanceId.getInstance().instanceId.await()
+
+                Timber.debug { "Found a token so proceed to subscribe the topic" }
+
+                topicsToBeSubscribed.forEach { topicName ->
+                    FirebaseMessaging.getInstance().subscribeToTopic(topicName).await()
+
+                    Timber.debug { "Subscribed $topicName successfully" }
+                }
+
+                topicsToBeUnsubscribed.forEach { topicName ->
+                    FirebaseMessaging.getInstance().unsubscribeFromTopic(topicName).await()
+
+                    Timber.debug { "Unsubscribed $topicName successfully" }
+                }
+            }
+        } catch (th: Throwable) {
+            Timber.error(th)
+
+            if (inputData.isValid) {
```

Comment:
I guess this is needless. *else* branch of this if statement is a dead flow because of https://github.com/DroidKaigi/conference-app-2019/pull/769/files#diff-f8cec374dddc9e3af5a74cf8d7bea514R27